mlir/include/mlir/Dialect/Vector/IR/VectorOps.td

```diff
@@ -207,6 +207,39 @@ def Vector_ContractionOp :
               .template getAsValueRange<IteratorTypeAttr, IteratorType>();
       return {range.begin(), range.end()};
     }
+
+    //===------------------------------------------------------------------===//
+    // The code below is shared with LinalgStructuredInterface.
+    // vector.contract is really a linalg.generic on vectors without region.
```

Comment:
If this is shared, let's try to identify a better location for it outside of the Vector dialect.

Also, could we avoid referring specifically to `Linalg` in generic `Vector` documentation? It was pointed out to me recently that not all users interact with `Vector` through `Linalg`, so keeping the documentation relevant to a broader audience would be helpful.